pkg/services/folder/folderimpl/folder.go

```diff
@@ -763,12 +769,18 @@ func (s *Service) BuildSaveDashboardCommand(ctx context.Context, dto *dashboards
 		}
 	}
 
+	namespaceID, userIDstr := dto.User.GetNamespacedID()
+	userID, err := identity.IntIdentifier(namespaceID, userIDstr)
```

Comment:
check namespace, if API key set user ID to 0 if you want to keep legacy support for API keys.